index.html

```diff
@@ -0,0 +1,464 @@
+        </p>
+        <p>
+          The steps to <dfn>launch a web app</dfn> are given by the following algorithm and takes a {{LaunchParams}} |params:LaunchParams|.
+        </p>
+        <ol class="algorithm">
+          <li>If |params| is null, set |params| to a new {{LaunchParams}} with {{LaunchParams/targetURL}} set to [=manifest/start_url=].
+          </li>
+          <li>Set |client| to the result of running the steps to [=prepare a web app launch client=] passing [=manifest/route_to=] and |params|.{{LaunchParams/targetURL}}.
+          </li>
+          <li>Append |params| to the [=unconsumed launch params=] of the |client|'s document's {{Window.LaunchQueue}}.
+          </li>
+          <li>If the [=assigned launch consumer=] |consumer| is set:
+            <ol>
+              <li>[=list/For each=] |launch_params:LaunchParams| of [=unconsumed launch params=]:
+                <ol>
+                  <li>Invoke |consumer| with |launch_params|.
+                  </li>
+                </ol>
+              </li>
+              <li>Set [=unconsumed launch params=] to the empty list.
+              </li>
+            </ol>
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h2>
+          Choosing a [=Web App Launch Client=]
+        </h2>
+        <p>
+          A <dfn>web app launch client</dfn> is a [=service worker client/window client=] associated with the web app.
+        </p>
+        <p class="note">
+          The exact form of this association is up to the user agent e.g. a dedicated app window separated in its UI presentation from general hyperlink browsing.
+        </p>
+        <p>
+          To <dfn>prepare a web app launch client</dfn>, given |route to| and |target URL:URL|:
+        </p>
+        <ol class="algorithm">
+          <li>
+            <p>Switching on |route to|, run the following substeps:</p>
+            <dl class="switch">
+              <dt>[=route to/auto=]</code>
+              <dd>
+                <ol class="algorithm">
+                  <li>Let |route_to| be either [=route to/new-client=] or [=route to/existing-client-navigate=] according to the user agent's decision for which is most appropriate.
+                  </li>
+                  <li>Return the result of the steps to [=prepare a web app launch client=] passing |route to| and |target URL|.
+                  </li>
+                </ol>
+              </dd>
+              <dt>[=route to/new-client=]</code>
+              <dd>
+                <ol class="algorithm">
+                  <li>Let |client| be the result of <a href="https://html.spec.whatwg.org/multipage/browsers.html#creating-a-new-top-level-browsing-context">creating a new top-level browsing context</a> presented in accordance with [=manifest/display=].
```

Comment:
Similar to how I didn't like you "magically" getting route_to out of the manifest above, you can't magically get display here. Make "prepare a web app launch client" take a processed manifest, as I said above, and then you can use "_manifest_["display"]".

Actually, probably the way to phrase this in accordance with the manifest spec (invoking the terminology established in [§1.17.4 Applying the manifest](https://www.w3.org/TR/appmanifest/#applying) would be:

"Let _client_ be the result of [creating a new top-level browsing context](https://html.spec.whatwg.org/multipage/browsers.html#creating-a-new-top-level-browsing-context) and [applying](https://www.w3.org/TR/appmanifest/#dfn-applied) _manifest_ to it (creating an [application context](https://www.w3.org/TR/appmanifest/#dfn-application-context))." (You could leave out the parenthetical if you find it awkward; it is implied by the definition of application context.)

You don't need to say anything about using `display`, since that's implied by [applying](https://www.w3.org/TR/appmanifest/#dfn-applied).

Reply:
Done.